### PR TITLE
[HGI 404] update pinned version of sentry to 7.45.0

### PR DESCRIPTION
**Pull request by AnkitUnthinkable**
**What does this PR do?**

In this PR includes

1. sentry update from version 5.12.1 to 7.45.0 (latest).
2. We are using pluggable integrations RewriteFrame which is also getting updated from 5.12.1 to 7.45.0 (latest). 
3. Sentry options whitelistURLs and blacklistURLs are changed to allowURLs and denyURLs. 
4. Sentry options serverName and maxMessageLength is no longer available. 

Testing Docs:  https://paper.dropbox.com/doc/Sentry-version-upgrade-v5.12.1-to-v7.45.0-Testing-Docs.--B2Jx3D2y_~_YtsXrYJ1W1sssAQ-hdyjHRh7cPrSg7AoMg66Y

**Testing**

Testing completed successfully in localhost

Jira tickets: 
https://segment.atlassian.net/browse/HGI-404?atlOrigin=eyJpIjoiMGVmZDgwYWExMWI2NGNjNThmZjhmZTc3ZWYwM2FjNzMiLCJwIjoiaiJ9

PREVIOUS VERSION

Screenshot 2023-03-28 at 5 22 16 AM

Current Version

Screenshot 2023-03-28 at 5 21 50 AM

### CODE REVIEW

**Review by varadarajan-tw — COMMENTED** [review]
Can we ensure test cases are updated? Also, since this is a major version upgrade, can we confirm all fields and settings are working as expected?.

File: integrations/sentry/lib/index.js [hunk 1]
```diff
@@ -16,24 +16,22 @@ var Sentry = (module.exports = integration('Sentry')
   .global('Sentry')
   .option('config', '')
   .option('environment', null)
-  .option('serverName', null)
   .option('release', null)
   .option('ignoreErrors', []) // still exists, but not documented on Sentry's website
   .option('ignoreUrls', [])
-  .option('whitelistUrls', [])
+  .option('allowUrls', [])
```
[the remark this answers is not in the document]
Comment (varadarajan-tw) [reply integrations/sentry/lib/index.js:22]: @AnkitUnthinkable - I don't see this change reverted yet? Are you missing a commit?

File: integrations/sentry/lib/index.js [hunk 2]
```diff
@@ -16,24 +16,22 @@ var Sentry = (module.exports = integration('Sentry')
...
   .option('includePaths', []) // maps to Sentry.Integrations.RewriteFrames plugin
-  .option('maxMessageLength', null) // deprecated
   .option('logger', null)
   .option('customVersionProperty', null)
   .option('debug', false)
   .tag(
     'sentry',
-    '<script src="https://browser.sentry-cdn.com/5.12.1/bundle.min.js" integrity="sha384-y+an4eARFKvjzOivf/Z7JtMJhaN6b+lLQ5oFbBbUwZNNVir39cYtkjW1r6Xjbxg3" crossorigin="anonymous"></script>'
+    '<script src="https://browser.sentry-cdn.com/7.45.0/bundle.min.js" integrity="sha384-eB2/mQAt3oY62hGYFXiPg18greyp8WT/GvKHlsvdYbvSxBRGEhBqEX8L7giHxzvp" crossorigin="anonymous"></script>'
```
[the remark this answers is not in the document]
Comment (varadarajan-tw) [reply integrations/sentry/lib/index.js:29]: @AnkitUnthinkable - Can you update the PR description to list all changes from 5.12.1 to 7.45.0?

**Comment by varadarajan-tw** [comment]
Testing Doc - https://paper.dropbox.com/doc/Sentry-version-upgrade-v5.12.1-to-v7.45.0-Testing-Docs.--B2Jx3D2y_~_YtsXrYJ1W1sssAQ-hdyjHRh7cPrSg7AoMg66Y

**Review by varadarajan-tw — COMMENTED** [review]
Left few more questions and comments

File: integrations/sentry/lib/index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (AnkitUnthinkable) [reply integrations/sentry/lib/index.js:29]: Sure @varadarajan-tw , Also @marinhero Added Testing docs please review it once. Thanks

**Comment by varadarajan-tw** [comment]
The changes look good! 
We are deprecating two settings
1. ServerName - It was never supported for browser SDK by Sentry
2. MaxMessageLength - It was already unsupported in the previous version this destination was on.

@brennan - Any thoughts on whether to delete such deprecated settings from [partner portal](https://app.segment.com/partner-portal/integration/sentry/settings) or just add deprecation warning and leave it as such?
Note: There is no cloud mode destination for Sentry.

**Review by varadarajan-tw — APPROVED** [review]

**Comment by brennan** [comment]
Hey @varadarajan-tw - I'm a fan of deprecating settings- as long as we test first in stage b/c we don't have a specific process for deprecating or removing settings other than simply deleting from partner portal. Simply deleting them from partner portal _should_ be ok, so long as we're not doing any operations on the setting in `integrations-consumer` or the monoservice, e.g. `ServerName.length` or something.

tl;dr - it's ok to leave them, but _if there is time_, it may be a good opportunity to document a process for removing deprecated settings. If so, then we should document and remove these.
